### PR TITLE
feat(core): add page group and display properties

### PR DESCRIPTION
**Pull request by JimmFly**
close TOV-23

https://github.com/toeverything/AFFiNE/assets/102217452/c05474de-b73c-40ab-9f18-cc43bb9fd828

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#6228** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6228?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/page-list/scoped-atoms.tsx [hunk 1]
```diff
@@ -191,14 +191,6 @@ export const groupsAtom = atom(get => {
 
   if (groupBy === false) {
     groupBy = undefined;
-  } else if (groupBy === undefined) {
```
Comment (pengx17) [inline packages/frontend/core/src/components/page-list/scoped-atoms.tsx:194]: I forgot why I added this else branch. It seems groupBy should never be false now. Maybe you can remove `false` from groupBy's type

File: packages/frontend/core/src/pages/workspace/all-page/all-page-header.tsx [hunk 2]
```diff
@@ -33,15 +167,35 @@ export const AllPageHeader = ({
...
+              iconPosition="end"
+              icon={
+                <ArrowDownSmallIcon className={styles.arrowDownSmallIcon} />
+              }
+              className={styles.headerDisplayButton}
+            >
+              {t['com.affine.page.display']()}
+            </Button>
+          </Menu>
+        </>
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/all-page/all-page-header.tsx]: Move the menu & related data logic out of AllPageHeader

**Review by pengx17 — COMMENTED** [review]
Please add some new e2e tests for grouping &  display properties

**Review by pengx17 — APPROVED** [review]

**Pull request merged**